lib/tapioca/compilers/dsl/url_helpers.rb

```diff
@@ -0,0 +1,82 @@
+# typed: true
+# frozen_string_literal: true
+
+require "parlour"
+
+begin
+  require "rails"
+rescue LoadError
+  return
+end
+
+module Tapioca
+  module Compilers
+    module Dsl
+      class UrlHelpers < Base
+        extend T::Sig
+
+        sig { override.params(root: Parlour::RbiGenerator::Namespace, constant: T.class_of(Object)).void }
+        def decorate(root, constant)
+          named_routes = Rails.application.routes.named_routes
+          path_helper_methods = named_routes.path_helpers_module.instance_methods(false)
+          url_helper_methods = named_routes.url_helpers_module.instance_methods(false)
+
+          if constant == GeneratedPathHelpersModule
+            root.create_module("GeneratedPathHelpersModule") do |mod|
+              mod.create_include("ActionDispatch::Routing::UrlFor")
+              mod.create_include("ActionDispatch::Routing::PolymorphicRoutes")
+
+              path_helper_methods.each do |method|
+                mod.create_method(
+                  method.to_s,
+                  parameters: [Parlour::RbiGenerator::Parameter.new("*args", type: "T.untyped")],
+                  return_type: "String"
+                )
+              end
+            end
```

Comment:
Can we extract this to a method? That way we can move the `path_helper_methods` inside that method and add a guard clause to `return if path_helper_methods.empty?` so that we never even generate a file for this if there are no methods.